### PR TITLE
Update FastAPI icon

### LINKED ISSUE #10662: Update outdated FastAPI icon
### Brand Name

FastAPI

### Official Resources for Icon and Color

https://fastapi.tiangolo.com/img/logo-margin/logo-teal.png

### Additional Comments

The icon is licensed under the MIT license.

### PR DESCRIPTION
**Pull request by uncenter**
image

**Issue:** closes #10662.

**Popularity metric:** N/A.

### Checklist

- [x] I updated the JSON data in `_data/simple-icons.json`
- [x] I optimized the icon with SVGO or SVGOMG
- [x] The SVG `viewbox` is `0 0 24 24`

### Description

I use https://github.com/tiangolo/fastapi/blob/master/docs/en/docs/img/icon-white.svg as the source for the new logo and updated the permalink to the file. The hex color seemed off so I updated it based on the `--md-primary-fg-color` CSS variable on their website.

### CODE REVIEW

**Comment by uncenter** [comment]
Apologies, looks like the existing hex is correct (https://github.com/tiangolo/fastapi/blob/ffb4f77a11f83132b521ba0aac6c95792c19e797/docs/en/docs/img/logo-teal.svg#L34).

File: icons/fastapi.svg [hunk 1]
```diff
@@ -1 +1 @@
-<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>FastAPI</title><path d="M12 0C5.375 0 0 5.375 0 12c0 6.627 5.375 12 12 12 6.626 0 12-5.373 12-12 0-6.625-5.373-12-12-12zm-.624 21.62v-7.528H7.19L13.203 2.38v7.528h4.029L11.376 21.62z"/></svg>
\ No newline at end of file
+<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>FastAPI</title><path d="M12.004.03A12.004 11.966 0 0 0 0 11.996 12.004 11.966 0 0 0 11.996 23.97 12.004 11.966 0 0 0 24 12 12.004 11.966 0 0 0 12.008.03m-.831 5.42h7.555l-7.589 5.329h5.185L5.28 18.546l2.4-5.329 1.096-2.43Z"/></svg>
```
Comment (service-paradis) [inline icons/fastapi.svg:1]: I'm seeing some differences in the inner shape (yours in black). Ex.
image

You could probably be able to remove safely 2 nodes in the straight line:
image

**Review by service-paradis — CHANGES_REQUESTED** [review]
Thank you @uncenter! I left a few comments if you can take a look!

File: icons/fastapi.svg (hunk 1 above)
Comment (uncenter) [reply icons/fastapi.svg]: How does it look now? I've overlayed it as you did and I'm no longer seeing any issues.

Comment (uncenter) [reply icons/fastapi.svg]: Screenshot 2024-03-18 at 11 55 24 (Inkscape)

Hmm actually I'm seeing one little spot there. I'll try to manually adjust now...

Comment (uncenter) [reply icons/fastapi.svg]: That should do it!

**Review by service-paradis — CHANGES_REQUESTED** [review]

File: icons/fastapi.svg [hunk 2]
```diff
@@ -1 +1 @@
-<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>FastAPI</title><path d="M12 0C5.375 0 0 5.375 0 12c0 6.627 5.375 12 12 12 6.626 0 12-5.373 12-12 0-6.625-5.373-12-12-12zm-.624 21.62v-7.528H7.19L13.203 2.38v7.528h4.029L11.376 21.62z"/></svg>
\ No newline at end of file
+<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>FastAPI</title><path d="M12.002.03C5.374.03 0 5.388 0 11.997c-.005 6.61 5.367 11.971 11.998 11.974 6.63 0 12.004-5.36 12.002-11.97C24.003 5.395 18.634.036 12.007.03m-.836 5.425h7.55l-7.577 5.328h5.183l-11.048 7.76Z"/></svg>
```
[the remark this answers is not in the document]
Comment (uncenter) [reply icons/fastapi.svg:1]: I'm seeing the same. Working on it but it keeps being just off xD.

[the remark this answers is not in the document]
Comment (uncenter) [reply icons/fastapi.svg]: Updated it again. It seems like there are some extremely tiny difference with the outer circle still though, so I might just leave this one up to one of you guys. It doesn't help that Inkscape has got into the habit of crashing when I try to bring in the original SVG for comparison! 😭

**Comment by full2null** [comment]
LGTM

File: icons/fastapi.svg
Comment (adamrusted) [inline icons/fastapi.svg:1]: I'm seeing a slight difference between your PR in red, and the source in white.
Might just be an optimisation thing, would you mind taking a look?
image

**Review by adamrusted — CHANGES_REQUESTED** [review]

File: icons/fastapi.svg
Comment (uncenter) [reply icons/fastapi.svg:1]: Ah, good catch. I've redone the icon and upped the precision during optimization, could you have a look now?

Comment (adamrusted) [reply icons/fastapi.svg:1]: Path itself is a perfect match now - though you've re-introduced the extra points in the 'f' shape. Could you trim those out?
image

Comment (adamrusted) [reply icons/fastapi.svg:1]: We're back to [this](https://github.com/simple-icons/simple-icons/pull/10663/files/ed6c497b067622d2f13341d5708473ee156bd808..3faf95a16f41174872b5e0585654ff82a71e3506#r1605798062) now I'm afraid...

Comment (uncenter) [reply icons/fastapi.svg:1]: Damn it! I'll try one more time.

Comment (uncenter) [reply icons/fastapi.svg:1]: Alright, pushed a change. No extra nodes and no differences on my end.

**Review by adamrusted — APPROVED** [review]

**Pull request merged**